src/Type/MixedType.php

```diff
@@ -408,6 +408,22 @@ public function traverse(callable $cb): Type
 
 	public function isArray(): TrinaryLogic
 	{
+		if ($this->subtractedType !== null) {
+			$arrayType = new ArrayType(new MixedType(), new MixedType());
+
+			if ($arrayType->isSuperTypeOf($this->subtractedType)->yes()) {
```

Comment:
You need to switch this around.


No other code is needed for this to work, no instanceof UnionType etc.